ksqldb-parser/src/test/java/io/confluent/ksql/parser/AstBuilderTest.java

```diff
@@ -393,7 +395,85 @@ public void shouldNotBuildLambdaFunctionNotLastArguments() {
     );
 
     // Then:
-    assertThat(e.getMessage(), containsString("mismatched input '=>' expecting {',', ')'}"));
+    assertThat(e.getMessage(), containsString("no viable alternative at input 'TRANSFORM_ARRAY(X =>'"));
+  }
+
+  @Test
+  public void shouldBuildIntervalExpression() {
+    // Given:
+    final SingleStatementContext stmt = givenQuery("SELECT TIMESTAMPADD(Col6, 5 HOURS) FROM TEST1;");
+
+    // When:
+    final Query result = (Query) builder.buildStatement(stmt);
+
+    // Then:
+    assertThat(result.getSelect(), is(new Select(ImmutableList.of(
+        new SingleColumn(
+            new FunctionCall(
+                FunctionName.of("TIMESTAMPADD"),
+                ImmutableList.of(
+                    column("COL6"),
+                    new IntervalExpression(
+                        new IntegerLiteral(5),
+                        TimeUnit.HOURS
+                    )
+                )
+            ),
+            Optional.empty())
+    ))));
+  }
+
+  @Test
+  public void shouldBuildMultipleIntervalExpression() {
+    // Given:
+    final SingleStatementContext stmt = givenQuery("SELECT TIMESTAMPADD(Col6, 5 HOURS, 6 MINUTES) FROM TEST1;");
+
+    // When:
+    final Query result = (Query) builder.buildStatement(stmt);
+
+    // Then:
+    assertThat(result.getSelect(), is(new Select(ImmutableList.of(
+        new SingleColumn(
+            new FunctionCall(
+                FunctionName.of("TIMESTAMPADD"),
+                ImmutableList.of(
+                    column("COL6"),
+                    new IntervalExpression(
+                        new IntegerLiteral(5),
+                        TimeUnit.HOURS
+                    ),
+                    new IntervalExpression(
+                        new IntegerLiteral(6),
+                        TimeUnit.MINUTES
+                    )
+                )
+            ),
+            Optional.empty())
+    ))));
+  }
+
+  @Test
+  public void shouldNotBuildIntervalExpressionNotLastArgument() {
+    // Given:
+    final Exception e = assertThrows(
+        ParseFailedException.class,
+        () -> givenQuery("SELECT TIMESTAMPADD(5 HOURS, Col6) FROM TEST1;")
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString("no viable alternative at input 'TIMESTAMPADD(5 HOURS"));
+  }
+
+  @Test
+  public void shouldNotBuildLambdaAndInterval() {
+    // Given:
+    final Exception e = assertThrows(
+        ParseFailedException.class,
+        () -> givenQuery("SELECT TRANSFORM_ARRAY(Col4, X => X + 5, 5 HOURS) FROM TEST1;")
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString("mismatched input '5' expecting {"));
```

Comment:
Any way to change the error message? Users may get confused about which `5` parameter is.